Nos/Views/UNS/UNSErrorView.swift

```diff
@@ -0,0 +1,87 @@
+//
+//  UNSErrorView.swift
+//  Nos
+//
+//  Created by Matthew Lorentz on 10/13/23.
+//
+
+import SwiftUI
+import Dependencies
+import Logger
+
+struct UNSErrorView: View {
+    
+    @ObservedObject var controller: UNSWizardController
+    @Dependency(\.analytics) var analytics
+    @Dependency(\.unsAPI) var api
+    
+    var body: some View {
+        NavigationView {
+            VStack {
+                ScrollView {
+                    UNSStepImage { Image.unsOTP.offset(x: 7, y: 5) }
+                        .padding(40)
+                        .padding(.top, 50)
+                    
+                    PlainText(.oops)
+                        .font(.clarityTitle)
+                        .multilineTextAlignment(.center)
+                        .foregroundColor(.primaryTxt)
+                        .shadow(radius: 1, y: 1)
+                        .padding(.top, 20)
+                        .padding(.bottom, 3)
+                    
+                    PlainText(.anErrorOccurred)
+                        .font(.clarityTitle)
+                        .multilineTextAlignment(.center)
+                        .foregroundColor(.primaryTxt)
+                        .shadow(radius: 1, y: 1)
+                        .padding(.bottom, 20)
+                    
+                    Text(.tryAgainOrContactSupport)
+                        .fontWeight(.medium)
+                        .multilineTextAlignment(.center)
+                        .foregroundColor(.secondaryText)
+                        .padding(.vertical, 17)
+                        .padding(.horizontal, 20)
+                        .shadow(radius: 1, y: 1)
+                        .padding(.bottom, 20)
+                }
+                
+                Spacer()
+                
+                BigActionButton(title: .goBack) {
+                    if api.accessToken != nil {
+                        do { 
+                            try await controller.navigateToChooseOrRegisterName()
+                        } catch {
+                            Log.optional(error)
+                            controller.state = .enterPhone
+                        }
+                    } else {
+                        controller.state = .enterPhone
+                    }
+                }
+                .padding(.bottom, 41)
+            }
+            .padding(.horizontal, 38)
+            .readabilityPadding()
+            .background(Color.appBg)
+            .onAppear {
+                switch controller.state {
+                case .error(let error):
+                    Log.optional(error, "UNSWizard encountered an error.")
+                    analytics.encounteredUNSError(error)
```

Comment:
It is a weird spot to send this analytics. Is it guaranteed to send just one error instance to sentry? Could it be better placed inside the controller code? Also, should some of them be sent to sentry as well (at least those we might want to detect in the production code to detect that UNS stopped working, because I think we won't detect them easily as we just set UNS once and will forget about it)

Reply:
Oh, I think most of our analytics are being sent in View code, so it didn't feel weird to put this in. I agree it's weird that we have to switch on the controller.state because really this view should never be presented unless the controller.state is .error, but I didn't create a model strict enough to guarantee that because I am rushing a little bit.

Good point about sending errors to Sentry. At some point we should probably set things up so that we have one function we can call that logs an error to the console and to Sentry, but for now I will just manually do it here.